lib/chai/utils/expectTypes.js

```diff
@@ -30,10 +30,11 @@ module.exports = function expectTypes(obj, types, ssfi) {
   flagMsg = flagMsg ? flagMsg + ': ' : '';
 
   obj = flag(obj, 'object');
+  // TODO: Fix - results in undefined when passed date
```

Comment:
It looks to me like `expectTypes` can only be used on assertion objects in order to make sure the target of the assertion (i.e., the value stored in the `object` flag) is one of the allowed types. It can't be used to check that arguments passed to an assertion are of the correct type; the `object` flag would be `undefined` regardless of the argument type. So you're doing the correct thing by checking the argument type manually.